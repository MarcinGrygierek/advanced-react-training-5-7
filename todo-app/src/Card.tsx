interface Props {
    as: 'div' | 'li';
    children: React.ReactNode;
    title: string;
    main?: boolean;
}

export const Card = ({as, children, title, main}: Props) => {
    const className = "card mt-3";

    const content = <div className='card-body'>
        {main ? <h1 className='card-title'>{title}</h1> : <h2 className='card-title'>{title}</h2>}
        {children}
    </div>

    if(as === 'div') {
        return <div className={className}>
            {content}
        </div>
    }
    return <li className={className}>
        {content}
    </li>
}
