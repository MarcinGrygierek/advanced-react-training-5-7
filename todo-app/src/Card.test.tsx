import * as renderer from 'react-test-renderer';
import { Card } from './Card';

describe('Card', () => {
    it('Card renders correctly as div', () => {
        const tree = renderer
            .create(
                <Card as='div' title='Title'>
                    Card description
                </Card>
            ).toJSON();
        expect(tree).toMatchSnapshot();
    })

    it('Card renders correctly as li', () => {
        const tree = renderer
            .create(
                <Card as='li' title='Title'>
                    Card description
                </Card>
            ).toJSON();
        expect(tree).toMatchSnapshot();
    })

    it('Card renders correctly as main', () => {
        const tree = renderer
            .create(
                <Card as='li' title='Title' main>
                    Card description
                </Card>
            ).toJSON();
        expect(tree).toMatchSnapshot();
    })
})