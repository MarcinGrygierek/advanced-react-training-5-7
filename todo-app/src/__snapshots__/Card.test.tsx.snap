// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Card renders correctly as div 1`] = `
<div
  className="card mt-3 accent"
>
  <div
    className="card-body"
  >
    <h2
      className="card-title"
    >
      Title
    </h2>
    Card description
  </div>
</div>
`;

exports[`Card Card renders correctly as li 1`] = `
<li
  className="card mt-3 accent"
>
  <div
    className="card-body"
  >
    <h2
      className="card-title"
    >
      Title
    </h2>
    Card description
  </div>
</li>
`;

exports[`Card Card renders correctly as main 1`] = `
<li
  className="card mt-3 main"
>
  <div
    className="card-body"
  >
    <h1
      className="card-title"
    >
      Title
    </h1>
    Card description
  </div>
</li>
`;
